owid/catalog/utils.py

```diff
@@ -20,7 +47,7 @@ def underscore_table(t: Table) -> Table:
     return t
 
 
-def validate_snake_case(name: Optional[str], object_name: str) -> None:
+def validate_underscore(name: Optional[str], object_name: str) -> None:
     """Raise error if name is not snake_case."""
     if name is not None and name != underscore(name):
```

Comment:
Can I suggest you do a regex match here, instead of using `underscore()`?

E.g.

```python
if not name or not re.match('^[a-z][a-z0-9_]+$', name):
     raise NameError(...)
```